### PR TITLE
Add requirements.txt to automated build

### PR DESCRIPTION
**Pull request by FreddieAkeroyd**
This allows specification of items needed for python venv on ISIS doc build server
 
See #899

It is possible to use a `requirements.txt` from a conda `environment.yml`
 
```
name: test-env
dependencies:
  - python>=3.5
  - anaconda
  - pip
  - pip:
    - -r file:requirements.txt
```

### CODE REVIEW

**Comment by FreddieAkeroyd** [comment]
have reduced `requirements.txt` to same 4 entries as `environment.yml` and it works fine

File: jenkins_build [hunk 1]
```diff
@@ -4,10 +4,16 @@ set -o errexit -x
...
 
-export PATH=/usr/local/bin:$PATH
-
-# use local python venv
-. /usr/local/python/bin/activate
+#export PATH=/usr/local/bin:$PATH
+
+## install local python virtual environment
+## based on local requirements.txt
+PYDIR=`pwd`/python
```
Comment (prjemian) [inline jenkins_build:11]: PYDIR=$(pwd)/python

Comment (prjemian) [reply jenkins_build:11]: This is a standard recommendation from Codacy's bash script analyses

Comment (FreddieAkeroyd) [reply jenkins_build:11]: does it show that I learnt my sh programming before that syntax was invented :-D

**Review by prjemian — COMMENTED** [review]
Lots of items in the requirements, many of which are probably required by a smaller number of packages we name here.  pip will examine the requirements of each package named in this file. This list may need to be pruned later as the packages we _actually_ need update their requirements.  If it works now, we can defer the pruning actions to a later issue.

**Comment by FreddieAkeroyd** [comment]
I pruned `requirements.txt` and it still seems to work on our local build server

**Review by prjemian — APPROVED** [review]
LGTM

**Pull request merged**